delivery_driver_stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -15,8 +15,19 @@ class StockPickingBatch(models.Model):
         store=True,
         domain="[('is_company', '=', False)]",
     )
+    user_id = fields.Many2one(
+        compute="_compute_user_id",
+        store=True,
+    )
 
     @api.depends("picking_ids.driver_id")
     def _compute_driver_ids(self):
         for batch in self:
             batch.driver_ids = batch.picking_ids.mapped("driver_id")
+
+    @api.depends("driver_ids")
+    def _compute_user_id(self):
+        for batch in self.filtered(lambda b: not b.is_wave):
+            users = batch.mapped("driver_ids.user_ids")
+            if users:
+                batch.user_id = users[0]
```

Comment:
So, the user is the first driver if several mentionned on all pickings of the batch ??

Reply:
Only one driver can drive a batch but as commented here: https://github.com/OCA/delivery-carrier/pull/775#issuecomment-1992038756 some times you want to join for get the maximun weight per truck so even initialy multidrivers should work, in this cases only one of them will make the delivery